### ISSUE TITLE
System Error When Loading Modules

### ISSUE THREAD

**Issue by Bitexirtz** [oqtane/oqtane.framework#525]
After installing from fresh checkout, I got an error when "Module Management" is clicked in "Admin Dashboard".

_packages is null.

![module1](https://user-images.githubusercontent.com/7827862/82635696-5ade3a80-9c33-11ea-81c6-98f50130554f.png)
![module2](https://user-images.githubusercontent.com/7827862/82635701-5ca7fe00-9c33-11ea-92b6-9b9d27f61d02.png)

**Comment by mikecasas** [comment]
Hi @Bitexirtz, I tried to reproduce the error and I couldn't. The packages object is just a list of packages from nuget. The code basically just calls this api ```https://azuresearch-usnc.nuget.org/query?q=tags:oqtane```.

Can you put a breakpoint in Server\Controllers\PackageController.cs and see what happens when the below route is hit.


```
 // GET: api/<controller>?tag=x
        [HttpGet]
        [Authorize(Roles = Constants.HostRole)]
        public async Task<IEnumerable<Package>> Get(string tag)
        {
            List<Package> packages = new List<Package>();

            using (var httpClient = new HttpClient())
            {
                CancellationToken token;
                var searchResult = await GetJson<SearchResult>(httpClient, "https://azuresearch-usnc.nuget.org/query?q=tags:oqtane", token);
                foreach(Data data in searchResult.Data)
                {
                    if (data.Tags.Contains(tag))
                    {
                        Package package = new Package();
                        package.PackageId = data.Id;
                        package.Name = data.Title;
                        package.Description = data.Description;
                        package.Owner = data.Authors[0];
                        package.Version = data.Version;
                        package.Downloads = data.TotalDownloads;
                        packages.Add(package);
                    }
                }
            }

            return packages;
        }

```

**Comment by Bitexirtz** [comment]
I think the cause of the problem is there is no internet connection. Cannot access json from azure.

**Comment by mikecasas** [comment]
OK, report back when you get connectivity.

**Comment by Bitexirtz** [comment]
No. I mean you can reproduce it by disconnecting your network connection. It should have no error if even no internet connection, right?

**Comment by mikecasas** [comment]
Sorry, I get what you mean now. Yes, when I shut off my WiFi, the http GET call crashes. I tried a couple of different ways to detect an Internet connection, but as long as you have a network interface it will report back true.

https://docs.microsoft.com/en-us/dotnet/api/system.net.networkinformation.networkinterface.getisnetworkavailable?redirectedfrom=MSDN&view=netcore-3.1#System_Net_NetworkInformation_NetworkInterface_GetIsNetworkAvailable

**Comment by sbwalker** [comment]
exception handling should be added to deal with this scenario

**Comment by sbwalker** [comment]
fixed in #557
